src/Attributes/Subgroup.php

```diff
@@ -8,15 +8,30 @@
 class Subgroup
 {
     public function __construct(
-        public string $name,
+        public mixed $name,
         public ?string $description = '',
     ){
     }
 
+    protected function getName(): string
```

Comment:
Ugh, I really hate these property accessor methods.

Considering making the next major PHP 8.4 only so I can do this with [property hooks](https://www.php.net/manual/en/language.oop5.property-hooks.php).